redux/longestStreakSlice.ts

```diff
@@ -0,0 +1,71 @@
+import { createSlice, PayloadAction, Slice } from "@reduxjs/toolkit";
+import { RootState } from "./rootReducer";
+import { BlockState } from "../components/math/longestStreak/MultiplicationBlock";
+import { GameBlockState } from "../pages/studentPortal/labs/multiplication/game";
+import { STAGE } from "../pages/studentPortal/labs/multiplication/game";
+
+export interface LongestStreakState {
+  stage: number;
+  blocks: GameBlockState[];
+  currentBlock: number;
+  isPlayerOneActive: boolean;
+  showPlayerOneName: string;
+  showPlayerTwoName: string;
+  isFilledBoard: boolean;
+  isGameReset: boolean;
+  isTwoPlayer: boolean;
+}
+
+const initialState: LongestStreakState = {
+  stage: 0,
+  blocks: [],
+  currentBlock: 0,
+  isPlayerOneActive: false,
+  showPlayerOneName: "Player 1",
+  showPlayerTwoName: "PLayer 2",
+  isFilledBoard: false,
+  isGameReset: false,
+  isTwoPlayer: false,
+
+};
+
+export const longestStreakSlice: Slice = createSlice({
+  name: "longestStreak",
+  initialState,
+  reducers: {
+    setStage: (state: LongestStreakState, action: PayloadAction<number>) => {
+        const stageOfGame = action.payload as number;
+
+        if (stageOfGame === STAGE.SET_RULES) {
```

Comment:
then you can remove this if/else statement and just do state.stage = stageOfGame